`in_type_checking_block` can crash if you haven't imported `TYPE_CHECKING`

### Bug description

```python3
if TYPE_CHECKING:  # don't import this!
    pass
else:
    import math
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
File "/Users/.../pylint/pylint/utils/ast_walker.py", line 96, in walk
    callback(astroid)
  File "/Users/.../pylint/pylint/checkers/variables.py", line 1329, in leave_module
    self._check_imports(not_consumed)
  File "/Users/.../pylint/pylint/checkers/variables.py", line 3071, in _check_imports
    if not in_type_checking_block(stmt):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../pylint/pylint/checkers/utils.py", line 1921, in in_type_checking_block
    maybe_import_from = ancestor.test.lookup(ancestor.test.name)[1][0]
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: tuple index out of range
```


### Expected behavior

```shell
************* Module a
a.py:1:3: E0602: Undefined variable 'TYPE_CHECKING' (undefined-variable)
a.py:4:4: W0611: Unused import math (unused-import)
```

### Pylint version

```shell
pylint 3.0.0a6
astroid 2.16.0dev0
Python 3.11.0
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8434.bugfix
+++ b/doc/whatsnew/fragments/8434.bugfix
@@ -1,0 +1,3 @@
+Fix a crash when ``TYPE_CHECKING`` is used without importing it.
+
+Closes #8434

--- a/pylint/checkers/utils.py
+++ b/pylint/checkers/utils.py
@@ -1918,7 +1918,10 @@ def in_type_checking_block(node: nodes.NodeNG) -> bool:
         if isinstance(ancestor.test, nodes.Name):
             if ancestor.test.name != "TYPE_CHECKING":
                 continue
-            maybe_import_from = ancestor.test.lookup(ancestor.test.name)[1][0]
+            lookup_result = ancestor.test.lookup(ancestor.test.name)[1]
+            if not lookup_result:
+                return False
+            maybe_import_from = lookup_result[0]
             if (
                 isinstance(maybe_import_from, nodes.ImportFrom)
                 and maybe_import_from.modname == "typing"

--- a/tests/checkers/unittest_utils.py
+++ b/tests/checkers/unittest_utils.py
@@ -446,6 +446,16 @@ def test_if_typing_guard() -> None:
     assert utils.is_typing_guard(code[3]) is False
 
 
+def test_in_type_checking_block() -> None:
+    code = astroid.extract_node(
+        """
+    if TYPE_CHECKING:  # don't import this!
+        import math  #@
+    """
+    )
+    assert utils.in_type_checking_block(code) is False
+
+
 def test_is_empty_literal() -> None:
     list_node = astroid.extract_node("a = []")
     assert utils.is_base_container(list_node.value)

